src/cpp/src/speculative_decoding/speculative_decoding_stateful.cpp

```diff
@@ -42,18 +43,6 @@ void update_perf_stat_by_infer_duration(ov::genai::RawPerfMetrics& raw_perf_coun
     raw_perf_counters.m_batch_sizes.emplace_back(num_generated_tokens);
 }
 
-void ensure_num_assistant_tokens_is_set(ov::genai::GenerationConfig& generation_config) {
```

Comment:
Please return this function back. Firstly, it defaults to `5` and not `4` candidates; secondly, this doesn't depend on the Eagle3 functionality.